### PR TITLE
fix(flags): Match properties in db irrespective of type

### PR DESCRIPTION
**Pull request by neilkakkar**
Instead of worrying about types, and not being able to set numeric types from the UI, make the matching algorithm type insensitive: it tests for both string and number types.

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: posthog/queries/base.py [hunk 1]
```diff
@@ -179,7 +179,14 @@ def empty_or_null_with_value_q(
 ) -> Q:
 
     if operator == "exact" or operator is None:
-        target_filter = lookup_q(f"{column}__{key}", Property._parse_value(value))
+        value_as_given = Property._parse_value(value)
+        value_as_coerced_to_number = Property._parse_value(value, convert_to_number=True)
+        if value_as_given == value_as_coerced_to_number:
```
Comment (liyiy) [inline posthog/queries/base.py:184]: What is this for again? Doesn't the else statement encompass this case too? 😮

Comment (neilkakkar) [reply posthog/queries/base.py:184]: Can add a comment, tell me more about what's confusing here?

It's just an optimisation to not have an OR statement between when both values are the same

Comment (liyiy) [reply posthog/queries/base.py:184]: oh but if the value is a number and there exists a record for it, won't it return before evaluating the second half of the `|` statement? Or it always evaluates both

Comment (neilkakkar) [reply posthog/queries/base.py:184]: SQL evaluation is not shortcircuited, the where clauses are unpredictable (i.e. the first where clause might not be the one that's evaluated first)

Comment (neilkakkar) [reply posthog/queries/base.py:184]: this Q object generates a SQL in the end, and that SQL is passed to the db, which then returns on appropriate match; the more concise we can make this SQL, the better.

Comment (liyiy) [reply posthog/queries/base.py:184]: TIL! thanks 🤗

**Review by liyiy — APPROVED** [review]

**Pull request merged**